---
title: "sc_rnaseq_seurat"
output: html_notebook
---
Outline:
1. Setup Seurat object
2. Pre-processing
3. Clustering


#Setup Seurat object

###Load packages
```{r}
install.packages("Seurat")
library(dplyr)
library(Seurat)
library(patchwork)
```

###Load data from 1k pbmc
```{r}
pbmc.data <- Read10X(data.dir = "~/Documents/GitHub/sc_rnaseq_10xpipeline/outs/filtered_feature_bc_matrix/")
pbmc <- CreateSeuratObject(counts = pbmc.data, project = "pbmc1k", min.cells = 3, min.features = 200)
pbmc
```

###Random pick few genes and check count matrix
```{r}
#row as genes and column as cells. below show 3 genes counts in first 30 cells
pbmc.data[c("CD3D", "TCL1A", "MS4A1"), 1:30]
```

#Pre-processing

###QC and selecting cells for further analysis
```{r}
# Show QC metrics for the first 5 cells
# first column = cellular barcodes
# nCount_RNA = number of UMIs per cell
# nFeature_RNA = number of genes per cell
# percent.mt = percentage of mitochondrials
head(pbmc@meta.data, 5)
pbmc[["percent.mt"]] <- PercentageFeatureSet(pbmc, pattern = "^MT-")

```

###Visualize QC metrics as a violin plot
```{r}
VlnPlot(pbmc, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
```

###Visualize feature-feature relationship as a Scatter plot 
```{r}
plot1 <- FeatureScatter(pbmc, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(pbmc, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
plot1 + plot2
```

###Filtering
```{r}
pbmc <- subset(pbmc, subset = nFeature_RNA > 200 & nFeature_RNA < 4500 & percent.mt < 25)
```

###Normalization
```{r}
pbmc <- NormalizeData(pbmc, normalization.method = "LogNormalize", scale.factor = 10000)
```

###Feature selection
```{r}
pbmc <- FindVariableFeatures(pbmc, selection.method = "vst", nfeatures = 2000)

# Identify the 10 most highly variable genes
top10 <- head(VariableFeatures(pbmc), 10)

# plot variable features with and without labels
plot1 <- VariableFeaturePlot(pbmc)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE)
plot2
```

###Scaling to normal expression [0,1]
```{r}
pbmc <- ScaleData(pbmc) #only scale on previous selected 2000 features
```

#Clustering

###PCA analysis
```{r}
pbmc <- RunPCA(pbmc, features = VariableFeatures(object = pbmc))
DimPlot(pbmc, reduction = "pca")
```

###Use K-nearest neighbor to cluster
```{r}
pbmc <- FindNeighbors(pbmc, dims = 1:10)
pbmc <- FindClusters(pbmc, resolution = 0.5)
head(Idents(pbmc), 5)
```

###Non-linear dimensional reduction (UMAP/tSNE)
```{r}
#reticulate::py_install(packages ='umap-learn')
pbmc <- RunUMAP(pbmc, dims = 1:10)
DimPlot(pbmc, reduction = "umap", label = TRUE)
#export
saveRDS(pbmc, file = "~/Documents/GitHub/sc_rnaseq_10xpipeline/pbmc_dim10.rds")
```

###Finding cluster markers (differentially expressed features)
```{r}
pbmc.markers <- FindAllMarkers(pbmc, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
pbmc.markers %>%
    group_by(cluster) %>%
    slice_max(n = 2, order_by = avg_log2FC)
```

###Assigning cell type identity to clusters (annotation)
```{r}
#manual match cluster markers/genes to cell type. ex. cluster0 has IL7R and TRBC1 which may belongs to Naive CD4+T cells. Below annotation is just an example.
new.cluster.ids <- c("Naive CD4 T", "CD14+ Mono", "Memory CD4 T", "B", "CD8 T", "FCGR3A+ Mono", "NK", "DC", "Platelet", "Mast cell", "Treg")
names(new.cluster.ids) <- levels(pbmc)
pbmc <- RenameIdents(pbmc, new.cluster.ids)
DimPlot(pbmc, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
#export
saveRDS(pbmc, file = "~/Documents/GitHub/sc_rnaseq_10xpipeline/pbmc_dim10_final.rds")
```

